pyproject.toml

```
@@ -1,9 +1,63 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "wheel",
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "django-allauth-2fa"
+dynamic = ["version"]
+description = "Adds two factor authentication to django-allauth"
+readme = "README.rst"
+license = "Apache-2.0"
+requires-python = ">=3.7"
+authors = [
+    { name = "Víðir Valberg Guðmundsson", email = "valberg@orn.li" },
+]
+
+keywords = [
+    "2fa",
+    "allauth",
+    "auth",
+    "authentication",
+    "django",
+    "factor",
+    "otp",
+    "two",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Environment :: Web Environment",
+    "Framework :: Django",
+    "Framework :: Django :: 3.2",
+    "Framework :: Django :: 4.0",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Internet",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+
+dependencies = [
+    "django-allauth>=0.41.0",
+    "django-otp>=1.0.0",
+    "django>=3.2",
+    "qrcode>=5.3",
+]
+
+[project.urls]
+Homepage = "https://github.com/valohai/django-allauth-2fa"
+
+[tool.hatch.version]
+path = "allauth_2fa/__init__.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/allauth_2fa",
```

I think we should also include the LICENSE

```suggestion
    "/allauth_2fa",
    "LICENSE",
```

---

I think `hatchling` includes it by default.

---

Ah cool, but as far as I can tell, from https://hatch.pypa.io/latest/plugins/builder/sdist/#default-file-selection, that only happens if we define  `license-files` (https://hatch.pypa.io/latest/config/metadata/#__tabbed_10_2)

---

Ah no that is source distribution